recipes/full_finetune_distributed.py

```diff
@@ -657,12 +705,18 @@ def train(self) -> None:
                 # Step with optimizer
                 if (idx + 1) % self._gradient_accumulation_steps == 0:
                     if self._clip_grad_norm is not None:
-                        grad_norm = torch.nn.utils.clip_grad_norm_(
-                            self._model.parameters(),
-                            max_norm=float(self._clip_grad_norm),
-                        )
-                    self._optimizer.step()
-                    self._optimizer.zero_grad(set_to_none=True)
+                        if self._optimizer_in_bwd:
+                            raise NotImplementedError(
+                                "Gradient clipping is not supported after optimizer-in-the-backward."
```

Comment:
`optimizer_in_backward` frees gradient information during `loss.backward`, could not get the correct `grad_norm`